test/test_hyperparameters.py

```diff
@@ -328,6 +326,176 @@ def test_normalfloat_to_integer(self):
+            str(f1_))
+
+        # test parameters that do not create a legit beta distribution
+        with self.assertRaises(ValueError):
+            BetaFloatHyperparameter("param", lower=-2, upper=2, alpha=-11, beta=5)
+            BetaFloatHyperparameter("param", lower=-2, upper=2, alpha=5, beta=-11)
+
+        # test parameters that do not yield a finite co-domain
+        with self.assertRaises(ValueError):
+            BetaFloatHyperparameter("param", lower=-2, upper=2, alpha=0.5, beta=11)
+            BetaFloatHyperparameter("param", lower=-2, upper=2, alpha=11, beta=0.5)
+
+        u1 = UniformFloatHyperparameter("param", lower=0.0, upper=1.0)
+        b1 = BetaFloatHyperparameter("param", lower=0.0, upper=1.0, alpha=3.0, beta=1.0)
+
+        # with identical domains, beta and uniform should sample the same points
+        self.assertEqual(u1.get_neighbors(0.5, rs=np.random.RandomState(42)),
+                         b1.get_neighbors(0.5, rs=np.random.RandomState(42)),
+                         )
+        self.assertEqual(f1.get_neighbors(0.5, rs=np.random.RandomState(42)), [
+                         0.8973713224089861, 0.38938855906305225,
+                         1.018150830480554, 1.7184238851264204])
+
+        b1_ext = BetaFloatHyperparameter("param", lower=-12.0, upper=12.0, alpha=3.0, beta=1.0)
+        self.assertEqual(b1_ext.get_neighbors(11.99, rs=np.random.RandomState(42)), [
+                         11.326331354378313, 10.866063801327988,
+                         10.866142606643933, 9.73652294751223])
+
+        b1_log = BetaFloatHyperparameter(
+            "param", lower=1.0, upper=1000.0, alpha=3.0, beta=1.0, log=True)
+        self.assertEqual(b1_log.get_neighbors(np.log(1000),
+                         rs=np.random.RandomState(42), transform=True), [
+                         826.1167337763767, 723.6156902737724,
+                         723.6321035060932, 522.7756713749886])
+        self.assertEqual(b1_log.get_neighbors(np.log(1),
+                         rs=np.random.RandomState(42), transform=True), [
+                         1.986225219010177, 2.4468825448117744,
+                         8.200076685608781, 8.861917227659887])
+
+        # Test attributes are accessible
+        self.assertEqual(f1.name, "param")
+        self.assertAlmostEqual(f1.alpha, 3.0)
+        self.assertAlmostEqual(f1.beta, 1.0)
+        self.assertAlmostEqual(f1.q, None)
+        self.assertEqual(f1.log, False)
+        self.assertAlmostEqual(f1.default_value, 2.0)
+        # beta parameters are not normalized (as normal parameters are not, either.)
+        self.assertAlmostEqual(f1.normalized_default_value, 2.0)
+
+        # Test copy
+        copy_f1 = copy.copy(f1)
+
+        self.assertEqual(copy_f1.name, f1.name)
+        self.assertEqual(copy_f1.alpha, f1.alpha)
+        self.assertEqual(copy_f1.beta, f1.beta)
+        self.assertEqual(copy_f1.default_value, f1.default_value)
```

Comment:
If you have a test for equality then you can rely on that :)
`self.assertEqual(f1, copy.copy(f1))`

Reply:
Yeah, that'd be shorter!